app/models/manageiq/providers/base_manager/event_catcher/runner.rb

```diff
@@ -138,6 +140,9 @@ def process_events(events)
 
   def do_work
     if @tid.nil? || !@tid.alive?
+      # Join the event monitor thread if it raised an exception
+      @tid.join(0.1) if @tid.status.nil? && @tid.stop?
```

Comment:
@Fryguy  Thanks!  I was dwelling on this logic over the weekend. I'm going to rework it a bit.